base/repl/REPLCompletions.jl

```diff
@@ -11,7 +11,11 @@ function completes_global(x, name)
 end
 
 function appendmacro!(syms, macros, needle, endchar)
-    append!(syms, s[2:end-sizeof(needle)]*endchar for s in filter(x -> endswith(x, needle), macros))
+    r = Regex("^.(.*)$needle\$")
```

Comment:
Perhaps `Regex("^.(.*)\\Q$needle\\E\$")` since you don't want special characters to have special interpretation?

Reply:
Sure :). I will fix after I am sure that CI goes through correcty (as it seems now will go through without errors).

Reply:
The CI looks fine now, failures are unrelated.